<a href="https://colab.research.google.com/github/Gnanaprakaasham/Breast_cancer_detection/blob/master/Breastcancer_detection_final.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Importing libraries**

In [5]:
import pandas as pd
import numpy as np

**Importing dataset**

The dataset used has no missing values and hence feature engineering is not required here.

In [6]:
data = pd.read_csv('breast_cancer.csv')

In [7]:
X = data.iloc[:,1:-1].values
y = data.iloc[:,-1].values

In [8]:
print(X)

[[ 5  1  1 ...  3  1  1]
 [ 5  4  4 ...  3  2  1]
 [ 3  1  1 ...  3  1  1]
 ...
 [ 5 10 10 ...  8 10  2]
 [ 4  8  6 ... 10  6  1]
 [ 4  8  8 ... 10  4  1]]


In [9]:
print(y)

[2 2 2 2 2 4 2 2 2 2 2 2 4 2 4 4 2 2 4 2 4 4 2 2 4 2 2 2 2 2 2 4 2 2 2 4 2
 4 4 4 4 4 4 2 4 2 2 4 4 4 4 4 4 4 4 4 4 4 4 2 4 4 2 4 2 4 4 2 2 4 2 4 4 2
 2 2 2 2 2 2 2 2 4 4 4 4 2 2 2 2 2 2 2 2 2 2 4 4 4 4 2 4 4 4 4 4 2 4 2 4 4
 4 2 2 2 4 2 2 2 2 4 4 4 2 4 2 4 2 2 2 4 2 2 2 2 2 2 2 2 4 2 2 4 2 2 4 2 4
 4 2 2 4 2 2 4 4 2 2 2 2 4 4 2 2 2 2 2 4 4 4 2 4 2 4 2 2 2 4 4 2 4 4 4 2 4
 4 2 2 2 2 2 2 2 2 4 4 2 2 2 4 4 2 2 2 4 4 2 4 4 4 2 2 4 2 2 4 4 4 4 2 4 4
 2 4 4 4 2 4 2 4 4 4 4 2 2 2 2 2 2 4 4 2 2 4 2 4 4 4 2 2 2 2 4 4 4 4 4 2 4
 4 4 2 4 2 4 4 2 2 2 2 4 2 2 4 4 4 4 4 2 4 4 2 2 4 4 2 2 4 4 2 4 2 4 4 2 2
 4 2 2 2 4 2 2 4 4 2 2 4 2 4 2 2 4 2 4 4 4 2 2 4 4 2 4 2 2 4 4 2 2 2 4 2 2
 2 4 4 2 2 2 4 2 2 4 4 4 4 4 4 2 2 2 2 4 4 2 2 2 2 2 2 2 2 2 2 2 2 2 4 2 2
 2 2 4 2 2 2 2 4 2 2 2 2 2 2 2 2 4 2 2 2 2 2 2 2 2 2 2 4 2 4 2 4 2 2 2 2 4
 2 2 2 4 2 4 2 2 2 2 2 2 2 4 4 2 2 2 4 2 2 2 2 2 2 2 2 4 2 2 2 4 2 4 4 4 2
 2 2 2 2 2 2 4 4 4 2 2 2 2 2 2 2 2 2 2 2 4 2 2 4 4 2 2 2 4 4 4 2 4 2 4 2 2
 2 2 2 2 2 2 2 2 2 2 4 2 

**Splitting dataset**

In [12]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.2,random_state= 0 )


**Logistic Regression on trainning set**

In [43]:
from sklearn.linear_model import LogisticRegression
lr = LogisticRegression()
lr.fit(X_train,y_train) 

LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=None, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)

In [44]:
y_pred_lr = lr.predict(X_test)

**Confusion matrix**

In [45]:
from sklearn.metrics import confusion_matrix,accuracy_score
cn = confusion_matrix(y_test,y_pred )

In [46]:
print(cn)

[[84  3]
 [ 3 47]]


In [47]:
from sklearn.model_selection import cross_val_score
cvrs = cross_val_score(lr,X_train,y = y_train,cv = 10)

In [48]:
print(cvrs)

[0.94545455 0.96363636 0.96363636 1.         0.94545455 1.
 0.96296296 0.96296296 0.98148148 0.94444444]


In [49]:
print("Accuracy: {:.2f}% ".format(cvrs.mean()*100))
print("Standard deviation : {:.2f}% ".format(cvrs.std()*100))

Accuracy: 96.70% 
Standard deviation : 1.97% 


In [53]:
accuracy_score(y_test,y_pred_lr)

0.9562043795620438

In [54]:
from xgboost import XGBClassifier
xgb = XGBClassifier()
xgb.fit(X_train,y_train)


XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, gamma=0,
              learning_rate=0.1, max_delta_step=0, max_depth=3,
              min_child_weight=1, missing=None, n_estimators=100, n_jobs=1,
              nthread=None, objective='binary:logistic', random_state=0,
              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,
              silent=None, subsample=1, verbosity=1)

In [55]:
y_pred_xb = xgb.predict(X_test)

In [59]:
cvrs = cross_val_score(xgb,X_train,y = y_train,cv = 10)

In [60]:
print(cvrs)

[0.90909091 0.96363636 0.98181818 0.98181818 0.94545455 0.98181818
 0.94444444 0.96296296 1.         0.96296296]


In [57]:
print("Accuracy: {:.2f}% ".format(cvrs.mean()*100))
print("Standard deviation : {:.2f}% ".format(cvrs.std()*100))

Accuracy: 96.34% 
Standard deviation : 2.45% 


In [58]:
accuracy_score(y_test,y_pred_xb)

0.9708029197080292

With the raise in accuracy by using XGBoost Classifier with 97.08% than with the logistic regression